File: android/app/src/main/java/com/djs66256/short_drama/feature/earn/model/EarnTaskPlayerResult.kt
package com.djs66256.short_drama.feature.earn.model

enum class EarnTaskPlayerResultReason(val wireValue: String) {
    PLAYBACK_ENDED("playback-ended"),
    USER_EXIT("user-exit"),
    BACKGROUNDED("backgrounded"),
    ERROR("error"),
    CONTAINER_RECREATED("container-recreated"),
}

data class EarnTaskPlayerResult(
    val taskId: String,
    val videoId: String,
    val completed: Boolean,
    val reason: EarnTaskPlayerResultReason,
    val source: String = EARN_SOURCE,
) {
    fun isValid(): Boolean {
        return source == EARN_SOURCE && taskId.isNotBlank() && videoId.isNotBlank()
    }
}

File: ios/ShortDrama/Sources/Domain/Entities/EarnTaskPlayerResult.swift
import Foundation

struct EarnTaskPlayerResult: Equatable, Sendable {
    enum Reason: String, Equatable, Sendable {
        case playbackEnded = "playback-ended"
        case userExit = "user-exit"
        case backgrounded = "backgrounded"
        case error
        case containerRecreated = "container-recreated"
    }

    let taskId: String
    let videoId: String
    let completed: Bool
    let reason: Reason
    let source: String

    init?(
        taskId: String,
        videoId: String,
        completed: Bool,
        reason: Reason,
        source: String = "earn"
    ) {
        let normalizedTaskId = taskId.trimmingCharacters(in: .whitespacesAndNewlines)
        let normalizedVideoId = videoId.trimmingCharacters(in: .whitespacesAndNewlines)
        let normalizedSource = source.trimmingCharacters(in: .whitespacesAndNewlines)

        guard !normalizedTaskId.isEmpty,
              !normalizedVideoId.isEmpty,
              normalizedSource == "earn" else {
            return nil
        }

        self.taskId = normalizedTaskId
        self.videoId = normalizedVideoId
        self.completed = completed
        self.reason = reason
        self.source = normalizedSource
    }
}
